### PR TITLE
Fix destructuring assignment in arrow functions without block

### LINKED ISSUE #8912: Destructuring array assignment fails in arrow functions without block
## Bug Report

**Current Behavior**
The `@babel/plugin-transform-destructuring` plugin throws an error on arrow functions without a block that contain a destructuring array assignment, e.g., `x = () => [a, b] = [1, 2]`.

**Input Code**
```bash
npm install @babel/cli@7.1.2 @babel/core@7.1.2 @babel/plugin-transform-destructuring@7.1.3
echo '
  const x = () => [a, b] = [1, 2];
  console.log(x());
' > test.js
echo '{ plugins: ["@babel/plugin-transform-destructuring"] }' > .babelrc
node test.js # works
node_modules/.bin/babel test.js # throws
```

**Expected behavior/code**
Babel transforms the above valid JavaScript code correctly.

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{ plugins: ["@babel/plugin-transform-destructuring"] }
```

**Environment**
- Babel version(s): @babel/cli@7.1.2 @babel/core@7.1.2 @babel/plugin-transform-destructuring@7.1.3
- Node/npm version: all
- OS: all
- Monorepo [e.g. yes/no/Lerna]: all
- How you are using Babel: all

**Possible Solution**

### PR DESCRIPTION
**Pull request by RubenVerborgh**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #8912
| Patch: Bug Fix?          | :+1:
| Major: Breaking Change?  | :-1:
| Minor: New Feature?      | :-1:
| Tests Added + Pass?      | :+1:
| Documentation PR Link    | n/a
| Any Dependency Changes?  | :-1:
| License                  | MIT

- In #8912, I reported that destructuring array assignment fails in arrow functions without a block.
- This pull requests adds fixtures for that case, as well as the neighboring case of arrow functions with block.
- It transforms such blockless functions into functions with a block and a return statement.

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
I would have changed the definition of `isBlockParent`, if it weren't autogenerated.

File: packages/babel-plugin-proposal-class-properties/test/fixtures/private-loose/non-block-arrow-func/output.mjs [hunk 1]
```diff
@@ -1,16 +1,14 @@
-export default (param => {
-  var _class, _temp, _props;
+var _class, _temp, _props;
```
[the remark this answers is not in the document]
Comment (RubenVerborgh) [reply packages/babel-plugin-proposal-class-properties/test/fixtures/private-loose/non-block-arrow-func/output.mjs]: You're absolutely right about scoping, let me look into that.

**Comment by RubenVerborgh** [comment]
@loganfsmyth Wrapping inside of `path.ensureBlock()` would mean making the implicit return value explicit. I'm proposing a more simple change that fixes the problem directly at the destructuring level, where we can easily insert a return statement.

**Comment by nicolo-ribaudo** [comment]
Thank you!

**Pull request merged**